app/controllers/reviews_controller.rb

```diff
@@ -28,24 +29,40 @@ def create
     end
   end
 
+  def edit
+    @release = Release.find(params[:release_id])
+    @review = Review.find(params[:id])
+    @release_genres = @release.genres
+  end
+
   def update
     if current_user
       @review = Review.find(params[:id])
-      @release = Release.find(params[:release_id])
-      if params[:commit] == "Upvote"
-        @review.votes += 1
-        @review.save
-      elsif params[:commit] == "Downvote"
-        @review.votes -= 1
-        @review.save
+      release = Release.find(@review.release)
+      if @review.update(review_params)
+        flash[:notice] = "Review successfully updated"
+      else
+        render :edit
```

Comment:
It may be worth including a flash notice here explaining that the review has NOT successfully updated.

Also, are you redirecting elsewhere REGARDLESS of whether the review has been successfully updated or not? Typically, we redirect upon a success (after issuing a flash message) and remain on the `edit` page in the case that the user needs to try again.